src/arrayMethodSort.js

```diff
@@ -4,8 +4,34 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = compare) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (compareFunction(this[i], this[j]) > 0) {
+          const curr = this[i];
+
+          this[i] = this[j];
+          this[j] = curr;
+        }
+      }
+    }
+
+    return this;
+  };
+}
+
+function compare(a, b) {
+  const firstValue = a.toString();
+  const secondValue = b.toString();
+
+  switch (true) {
```

Comment:
Avoid using switch(true), it's generally considered bad practice due to its lack of clarity. Use if/else statements for better readability.